au/magnitude.hh

```diff
@@ -131,7 +131,16 @@ using CommonMagnitudeT = typename CommonMagnitude<Ms...>::type;
 // Value based interface for Magnitude.
 
 static constexpr auto ONE = Magnitude<>{};
+
+#ifndef PI
+// Some users must work with frameworks that define `PI` as a macro.  Having a macro with this
+// easily collidable name is exceedingly unwise.  Nevertheless, that's not the users' fault, so we
+// accommodate those frameworks by omitting the definition of `PI` in this case.
+//
+// If you are stuck with such a framework, you can choose a different name that does not collide,
+// and reproduce the following line in your own system.
 static constexpr auto PI = Magnitude<Pi>{};
```

Comment:
I'm not sure if this is a perfect solution. It would depend on the order of includes. The order of headers is not great to reason about.

I don't have great ideas for alternative solutions 😓 
1. Use Google style guide for naming constants `kBlah`. It generally won't collide with macros, but it is ugly.
2. Rename `PI` to something like `AUPI`. It would be unique, but it's more verbose and farther from the simple Greek letter. Also, that is what namespaces are for 😄 
3. Define PI in another header. This would require more fine grained header management.

Reply:
Having slept on this, I'm leaning towards Option 4: just don't define `PI`.  Using really short all-caps names is a recipe for collision with poorly designed frameworks.  That's not to say we'll _never_ do it, but it seems wise to minimize how much we do it.  And in this case, we have a clear collision with the Arduino framework, which we'd really like to support, so I think our `PI` has to go.

I'll deprecate it in a follow on PR for ease of review.  I think we'll still want to retain the `#ifndef` guards until we remove it entirely (probably in 0.4.0).